includes/plugins/class-co-authors-plus.php

```diff
@@ -209,6 +209,9 @@ public static function user_profile_update_errors( $errors, $update, $user ) {
 
 			// Create user name from Display name.
 			$user->user_login = self::generate_username( $user->display_name );
+
+			// Create a placeholder email address to avoid any issues with empty emails.
+			$user->user_email = $user->user_login . '@example.com';
```

Comment:
The email in not required, but it _can_ be provided. It should be set to the dummy email only if it's not provided. Or simply inside the `! empty( $errors->errors['user_email'] )` conditional above.

Reply:
yes of course. Also I was not creating the placeholder email on user updates. Fixed now